Base/usr/share/man/man7/kernel_cmdline.md

```diff
@@ -0,0 +1,129 @@
+## Name
+
+Kernel cmdline - boot arguments to control system functionality
+
+## Description
+
+The kernel supports boot parameters to change some of its behavior. This
+functionality can be used to disable or enable features in the OS entirely, but
+can sometimes be used also to change a feature's behavior without disabling it
+entirely.
+
+Some of the boot parameters require a value after it. In that case, you will
+need to write them like this:
+
+```
+param1=value ...
+
+```
+For those that don't require, you should just write them without equals sign,
+like that:
+
+```
+param1 ...
+
+```
+
+## Where to put those parameters
+
+You can either add kernel boot parameters when you invoke `./run`:
+
+```
+./run qcmd param1 param2 ...
+
+```
+for Q35 machines, use this layout:
+
+```
+./run q35_cmd param1 param2 ...
+
+```
+
+You may wish to write the kernel boot parameters in a more consistent method.
+For that, you can modify the GRUB configuration files. Since we have 3 of them
+(one for each partition scheme), you should pick the right one for you, so if
+you build a virtual disk with a GPT partition scheme, you should edit
+`grub-gpt.cfg` file.
+
+## Kernel boot parameters list
+
+Since it's pretty much not convenient for beginners to have a travel in the
+`init.cpp` file, the list here is meant to aid users to find all the kernel boot
+parameters and their meaning.
+
+### `text_debug`
+
+This boot parameter requires no value after it. This boot parameter enables
+Shell mode during boot, therefore you won't boot into the graphical mode.
```

Comment:
"This argument instructs SystemServer " [so that the admin immediately knows what this affects!] "to spawn a shell instead of the usual services such as the WindowServer". Then a few words about text/graphical mode.

Reply:
Or actually I'm misremembering and this affects what the kernel spawns instead of the SystemServer. Then it should mention *that*.

Reply:
You are confused since we talked about this in the IRC channel.
What we wanted in a brief description is to make sure you can't spawn `/bin/Shell`, but to use something like runlevels, or something similar to systemd-targets.

But for now, this is not the case, and `text_debug` argument will spawn /bin/Shell.

Reply:
Yeah, that's why I thought that it was a userspace thing for a moment :)

Reply:
OK then, I'll expand the description a bit, hopefully it will be fine :)